internal/pkg/template/template_functions.go

```diff
@@ -0,0 +1,79 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package template
+
+import (
+	"fmt"
+	"regexp"
+	"strconv"
+	"strings"
+)
+
+const (
+	dashReplacement = "DASH"
+)
+
+// ReplaceDashes takes a CloudFormation logical ID, and
+// sanitizes it by removing "-" characters (not allowed)
+// and replacing them with "DASH" (allowed by CloudFormation but
+// not permitted in ecs-cli generated resource names).
+func ReplaceDashes(logicalID string) string {
```

Comment:
What do you think of ending all of these function names with `Func`
so that it reads `template.ReplaceDashesFunc` to denote that they are [template functions](https://golang.org/pkg/text/template/#hdr-Functions) vs. regular functions?